src/claimtriedb.cpp

```diff
@@ -0,0 +1,204 @@
+
+#include "claimtrie.h"
+#include "claimtriedb.h"
+
+#include <vector>
+#include <iterator>
+#include <boost/scoped_ptr.hpp>
+
+/// Concrete implementation of dirty queue
+template <typename K, typename V>
+struct CCMap : public CCBase
+{
+    CCMap() : data() {}
+    typedef std::vector<std::pair<K, V> > dataType;
```

Comment:
Why not an unordered_map? (or map if ordering matters). It appears from the method below that we commonly need to find by key.

Reply:
unordered_map is a hash_map, so we should provide a hash algorithm for some classes, like base_blob derived.

Reply:
I don't think the queues get to be that large, but perhaps someday it's worth measuring the performance.  As it is, you gain cache effects due to locality (I think this wins), but with a well defined hash and unordered_map you gain constant time retrieval.  This is not urgent for this PR and it's fine as-is.  I'd also advise (style) to just use std::find.std::find_if with a comparator if/when it's supported by the compiler (certainly after upstream merges).